bottlecap/src/bin/bottlecap/main.rs

```diff
@@ -75,11 +78,106 @@ use std::{
     sync::{Arc, Mutex},
     time::{Duration, Instant},
 };
-use tokio::sync::{mpsc::Sender, Mutex as TokioMutex};
+use tokio::{sync::mpsc::Sender, sync::Mutex as TokioMutex, task::JoinHandle};
 use tokio_util::sync::CancellationToken;
 use tracing::{debug, error};
 use tracing_subscriber::EnvFilter;
 
+#[allow(clippy::struct_field_names)]
+struct ShutdownHandles {
+    trace_flush_handles: FuturesOrdered<JoinHandle<Vec<SendData>>>,
+    log_flush_handles: FuturesOrdered<JoinHandle<Vec<reqwest::RequestBuilder>>>,
+    metric_flush_handles: FuturesOrdered<JoinHandle<(Vec<Series>, Vec<SketchPayload>)>>,
+}
+
+impl ShutdownHandles {
+    fn new() -> Self {
+        Self {
+            trace_flush_handles: FuturesOrdered::new(),
+            log_flush_handles: FuturesOrdered::new(),
+            metric_flush_handles: FuturesOrdered::new(),
+        }
+    }
+
+    async fn await_flush_handles(
+        &mut self,
+        logs_flusher: &LogsFlusher,
+        trace_flusher: &ServerlessTraceFlusher,
```

Comment:
Shouldn't this be an Arc?

Reply:
Nah at this point we've already cloned it